### ISSUE TITLE
pgsql: invalid byte sequence for encoding "UTF8": 0x00  when updating the last error message on repo table for corrupt repo

### ISSUE THREAD

**Issue by mike-r-mclaughlin** [sourcegraph/sourcegraph#20524]
- **Sourcegraph version:** 3.27.4

#### Steps to reproduce:
Unclear.  Requires a "corrupt" git repository.  Errors in gitserver at the time show:

```
fatal: unable to read `<git_sha>`
aborting due to possible repository corruption on the remote side
```

on `pgsql` pod we see:
```
2021-04-29 15:11:20.085 GMT [3272] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
```

We see a similar error when we try to do a `git clone` from the command line.

#### Expected behavior:

No error in the `pgsql` logs

#### Actual behavior:

Error is displayed in the `pgsql` logs:

```
2021-04-29 15:11:20.085 GMT [3272] ERROR:  invalid byte sequence for encoding "UTF8": **0x00**
```

**Comment by indradhanush** [comment]
@mike-r-mclaughlin I searched for this error and found a way to reproduce it locally:

```
localhost sourcegraph@sourcegraph=# select E'ab\0cd';
ERROR:  22021: invalid byte sequence for encoding "UTF8": 0x00
LOCATION:  report_invalid_encoding, mbutils.c:1593
Time: 0.127 ms
```

My understanding out of this is that either:

1. The customer has a non ascii character in their database 
2. Or the customer has the character `\0` in one of the columns

The most likely suspect is the `repo` table since the error appears to stem from it.

To try and isolate this, here's what I would suggest to the customer:

1. Try to run a query to search for the character `\0` in the `repo` table. The most common suspect is the `name` column. This query should do the job:

```
SELECT * FROM repo where name = 0x00;
```

This searches the `repo` table for repos with the name `0x00` since that's the byte sequence in the error message.

Also repeat this for the description field if the above does not return a result:

```
SELECT * FROM repo where description = 0x00;
```

2. If the above does not work, I would suggest the customer to take a database dump of the `repo` table
and grep for this character - `\0`. 

So something like this:

```
# Take the database dump for the repo table in the plain format to output a plain-text SQL script file
$ pg_dump -F plain -t repo > repo-dump.sql

# Grep for that character in the generated dump file
$ grep \\\\0 repo-dump.sql
```

I hope either of the above two approaches shows some results. And if it does, we will need to update that specific character with an empty string instead: `''`.

Additionally, during my research an I noticed an over arching theme about this issue. It seems users face this issue when importing mysql dumps into database. Is there any chance our customer was writing data to postgres from mysql at any point of time? This sounds outlandish but I thought maybe it wouldn't hurt to ask? :thinking:

**Comment by tsenart** [comment]
@indradhanush , I think this happens in a very specific place, which @ryanslade found out about before: https://github.com/sourcegraph/sourcegraph/issues/19610

**Comment by tsenart** [comment]
I think we just need to sanitize the error here: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/database/gitserver_repos.go?subtree=true#L180:4

**Comment by tsenart** [comment]
Fixed by #21334
